src/main/java/net/dancier/domain/Recommendation.java

```diff
@@ -0,0 +1,32 @@
+package net.dancier.domain;
+
+import net.dancier.domain.dance.Dancer;
+import net.dancier.domain.dance.School;
+
+public class Recommendation<T extends Recommendable> {
+
+    public enum Type {
+        DANCER,
+        SCHOOL,
+        WORKSHOP,
+        EVENING,
+        ETC
+    }
+
+    public Type type;
+    public T payload;
+
+    public static Recommendation of(Recommendable recommendable) {
```

Comment:
Add doc please. Document or better use use annotation to inform about null handling. Right now null will lead to ETC.